src/views/Nft/market/Collection/IndividualNFTPage/OneOfAKindNftPage/MainNFTCard.tsx

```diff
@@ -15,19 +15,20 @@ interface MainNFTCardProps {
   nft: NftToken
   isOwnNft: boolean
   nftIsProfilePic: boolean
+  onSuccess: () => void
 }
 
-const MainNFTCard: React.FC<MainNFTCardProps> = ({ nft, isOwnNft, nftIsProfilePic }) => {
+const MainNFTCard: React.FC<MainNFTCardProps> = ({ nft, isOwnNft, nftIsProfilePic, onSuccess }) => {
   const { t } = useTranslation()
   const bnbBusdPrice = useBNBBusdPrice()
 
   const currentAskPriceAsNumber = nft.marketData?.currentAskPrice ? parseFloat(nft.marketData.currentAskPrice) : 0
   const priceInUsd = multiplyPriceByAmount(bnbBusdPrice, currentAskPriceAsNumber)
   const [onPresentBuyModal] = useModal(<BuyModal nftToBuy={nft} />)
   const [onPresentSellModal] = useModal(
-    <SellModal variant={nft.marketData?.isTradable ? 'edit' : 'sell'} nftToSell={nft} />,
+    <SellModal variant={nft.marketData?.isTradable ? 'edit' : 'sell'} nftToSell={nft} onSuccess={onSuccess} />,
   )
-  const [onEditProfileModal] = useModal(<EditProfileModal />, false)
+  const [onEditProfileModal] = useModal(<EditProfileModal onSuccess={onSuccess} />, false)
```

Comment:
Could we use Context to avoid Prop Drilling ?

Reply:
We have state change events out of control of user such as incoming nft transfer, achievement (when claiming current ifo it rewards achievement automatically) or selling nft therefore we should not hold state to show user always correct data without forcing user to refresh webpage completely.

Reply:
Hmm. Don't get me wrong. I don't suggest global state. I think we just need to add Context where the parent is used. It would be a good call if we try to avoid Prop Drilling in this case.